### ISSUE TITLE
Consider switching to Xi Editor from Scintilla

### ISSUE THREAD

**Issue by emoon** [emoon/ProDBG#206]
https://github.com/google/xi-editor seems to have good momentum and might be a good fit as

"Separation into front-end and back-end modules. The front-end is responsible for presenting the user interface and drawing a screen full of text. The back-end (also known as “core”) holds the file buffers and is responsible for all potentially expensive editing operations."

Mean that it should be possible to have a imgui/bgfx renderer for this.

Still I would say it's a bit too early as it lacks any highlighting and such.

**Comment by gwihlidal** [comment]
This seems like it could be used for the syntax highlighting:

https://github.com/trishume/syntect

**Comment by emoon** [comment]
Yeah it was suggested here :) https://github.com/google/xi-editor/issues/23#issuecomment-224463856

**Comment by emoon** [comment]
This may still be interesting even if with the switch to Qt
